### PR TITLE
fix: new client mode vs singleton client mode

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Closes #

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -17,7 +17,11 @@ export function createApp(
     const config = createProxyConfig(options);
     const { logger } = config;
     logger.debug('Configuration:', config);
-    const client = unleashClient || createClient(config);
+    const client =
+        unleashClient ||
+        (options.clientMode === 'new'
+            ? createNewClient(config)
+            : createSingletonClient(config));
```
Comment (gastonfournier) [inline src/app.ts:24]: Should we have this as an option or just another method `createNonSingletonApp`? I'd like it to be `createSingletonApp` but that basically requires changing the current method behavior.

I'm ok with this approach I'm just wondering if you considered having a new method instead

File: src/create-client.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import { IProxyConfig } from './config';
 import Client from './client';
-import { initialize } from 'unleash-client';
+import { initialize, Unleash } from 'unleash-client';
 import Metrics from 'unleash-client/lib/metrics';
 import { defaultStrategies } from 'unleash-client/lib/strategy';
 
-export const createClient = (config: IProxyConfig): Client => {
+export const createSingletonClient = (config: IProxyConfig): Client => {
```
Comment (gastonfournier) [inline src/create-client.ts:7]: Because the createClient is an exported should we keep it? Maybe it can be an alias to `createSingletonClient` and it can be marked as deprecated.

**Review by gastonfournier — COMMENTED** [review]
This looks fine, but just want to clarify a few things

File: src/create-client.ts (hunk 2 above)
Comment (kwasniew) [reply src/create-client.ts:7]: createClient is only exported internally and used into the app.ts. We don't export it to proxy users. I created this create-client.ts a week ago to encapsulate dependency injection graph creation.

File: src/app.ts (hunk 1 above)
Comment (kwasniew) [reply src/app.ts:24]: I though that for our own code using proxy it'd be easiest to do:  unleashProxy.createApp({clientMode: 'new'}). Test it in production to find if there are no corner cases. Then once we're happy to remove clientMode config and switch to createNewClient for everyone and rename it back to createClient.

**Review by gastonfournier — APPROVED** [review]
Awesome! I think your comments makes sense :+1:
